sdk/clientcore/tools/annotation-processor/src/main/java/io/clientcore/tools/codegen/AnnotationProcessor.java

```diff
@@ -0,0 +1,196 @@
+
+        templateInput.setPackageName(packageName);
+        templateInput.setServiceInterfaceFQN(serviceInterfaceFQN);
+        templateInput.setServiceInterfaceShortName(serviceInterfaceShortName);
+        templateInput.setServiceInterfaceImplShortName(serviceInterfaceImplShortName);
+
+        // Read ServiceInterface.host() value from any enclosed interfaces
+        if (serviceInterface.getAnnotation(ServiceInterface.class) != null) {
+            templateInput.setHost(serviceInterface.getAnnotation(ServiceInterface.class).host());
+        }
+
+        // add all imports
+        addImports(templateInput);
+
+        // iterate through all methods in the interface, collecting all that are @HttpRequestInformation
+        // and then generating a method implementation for each one
+        templateInput.setHttpRequestContexts(serviceInterface.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                .filter(element -> element.getAnnotation(HttpRequestInformation.class) != null)
+                .map(ExecutableElement.class::cast)
+                .map(e -> createHttpRequestContext(e, templateInput))
+                .collect(Collectors.toList()));
+
+        // template input set UnexpectedResponseExceptionDetails
+        templateInput.setUnexpectedResponseExceptionDetails(serviceInterface.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                .filter(element -> element.getAnnotation(HttpRequestInformation.class) != null)
+                .map(ExecutableElement.class::cast)
+                .map(e -> e.getAnnotation(UnexpectedResponseExceptionDetail.class))
+                .collect(Collectors.toList()));
+
+        TemplateProcessor.getInstance().process(templateInput, processingEnv);
+
+        // format the generated code
+    }
+
+    private void addImports(TemplateInput templateInput) {
+        templateInput.addImport("io.clientcore.core.util.Context");
+        templateInput.addImport("io.clientcore.core.util.binarydata.BinaryData");
+        templateInput.addImport("io.clientcore.core.http.models.HttpHeaders");
+        templateInput.addImport("io.clientcore.core.http.pipeline.HttpPipeline");
+        templateInput.addImport("io.clientcore.core.http.models.HttpHeaderName");
+        templateInput.addImport("io.clientcore.core.http.models.HttpMethod");
+        templateInput.addImport("io.clientcore.core.http.models.HttpResponse");
+        templateInput.addImport("io.clientcore.core.http.models.HttpRequest");
+        templateInput.addImport("io.clientcore.core.http.models.Response");
+        templateInput.addImport("java.util.Map");
+        templateInput.addImport("java.util.HashMap");
+        templateInput.addImport("java.util.Arrays");
+        templateInput.addImport("java.lang.Void");
+        templateInput.addImport("java.util.List");
```

Comment:
Given we have a dependency on `io.clientcore:core` could we change all these to being something like `Content.class.getName()` as `core` is still in development and if these move it'll be obvious as compilation here will break.